packages/gafl-webapp-service/src/pages/multibuy/remove-licence/update-transaction.js

```diff
@@ -17,9 +17,11 @@ export default async request => {
   await request.cache().helpers.status.set(status)
   await request.cache().helpers.addressLookup.set(addressLookup)
 
-  const lastStatusPermission = status.permissions[status.permissions.length - 1]
-  await request.cache().helpers.status.set({ currentPermissionIdx: transaction.permissions.length - 1 })
-  await request.cache().helpers.status.setCurrentPermission(lastStatusPermission)
+  if (transaction.permissions.length > 0) {
+    const lastStatusPermission = status.permissions[status.permissions.length - 1]
+    await request.cache().helpers.status.set({ currentPermissionIdx: transaction.permissions.length - 1 })
+    await request.cache().helpers.status.setCurrentPermission(lastStatusPermission)
+  }
```

Comment:
I think you need an obverse test that this doesn't happen when transaction.permissions.length is zero. Also, are you testing for `request.cache().helpers.status.set` being called with the currentPermissionIdx? You need to do that, as well as testing that it isn't called when transaction.permissions.length is zero.

Reply:
I'm still not seeing an obverse test for when transaction.permissions.length is zero. Or for `request.cache().helpers.status.set`. I think we need to talk about test driven development, where you write your tests before you write any executing code.